logistics/scripts/modules/settings/settings-management.psm1

```diff
@@ -312,21 +321,40 @@ function New-DevelopmentAppSettings([hashtable] $Settings = @{ }) {
         $developmentSettingsForEngine = (Get-DefaultDevelopmentSettingsByEngine)[$Settings.ApiSettings.Engine]
         $developmentSettingsForEngine = Add-ApplicationNameToConnectionStrings $developmentSettingsForEngine $project
 
-        $newDevelopmentSettings = Merge-Hashtables $developmentSettingsByProject[$project], $developmentSettingsForEngine, $Settings
+        $newDevelopmentSettings = Merge-Hashtables $developmentSettingsByProject[$project], $developmentSettingsForEngine
 
         $newDevelopmentSettings = Add-TestHarnessSpecificAppSettings $newDevelopmentSettings $project
 
+        $newDevelopmentSettings = Merge-Hashtables $newDevelopmentSettings, (Get-CredentialSettingsByProject)[$project], $Settings
+
         $projectPath = Get-RepositoryResolvedPath $Project
 
         $newDevelopmentSettingsPath = Join-Path $projectPath "appsettings.development.json"
         New-JsonFile $newDevelopmentSettingsPath $newDevelopmentSettings -Overwrite
         $newSettingsFiles += $newDevelopmentSettingsPath
 
         $credentialSettings = Merge-Hashtables @{ }, (Get-CredentialSettingsByProject)[$project]
-        $newUserSettingsPath = Join-Path $projectPath "appsettings.user.json"
-        New-JsonFile $newUserSettingsPath $credentialSettings
-        $newSettingsFiles += $newUserSettingsPath
+        $userSecrets += ($credentialSettings | ConvertTo-Json -Depth 10 | dotnet user-secrets set --project $projectPath --id (Get-UserSecretsIdByProject).$project)
     }
 
-    return $newSettingsFiles
+    return ($newSettingsFiles, $userSecrets)
 }
+
+function Display-UserSecretsList() {
+    $developmentSettingsByProject = Get-DefaultDevelopmentSettingsByProject
+    $userSecrets =@{ }
+    foreach ($project in $developmentSettingsByProject.Keys) {       
+
+       $projectPath = Get-RepositoryResolvedPath $project       
+       $userSecretList= dotnet user-secrets list --project $projectPath --id (Get-UserSecretsIdByProject).$project |  Out-String  
+                
+       if($userSecretList -NotLike "*No secrets configured for this application*")
+       {
+           $userSecretList=ConvertFrom-StringData -StringData $userSecretList        
```

Comment:
I don't think `ConvertFrom-StringData` is going to give us the result that we need here. As we discussed the output from `dotnet user-secrets list` needs to be unflattened into a hashtable in order to be able to merge with the other settings. If we keep it flattened we need to do some pretty heavy modification to all the functions accessing the settings hashtable returned from `Get-DeploymentSettings`.

Reply:
I am able to unflatten now  merge into  Get-DeploymentSettings but I am getting error in conversion now